sys/include/net/gnrc/ppp/fsm.h

```diff
@@ -0,0 +1,291 @@
+        PPP_S_STOPPING, PPP_S_REQ_SENT, PPP_S_REQ_SENT, PPP_S_REQ_SENT, PPP_S_STOPPING },
+    { PPP_S_UNDEF, PPP_S_UNDEF, PPP_S_CLOSED, PPP_S_STOPPED, PPP_S_CLOSED, 
+        PPP_S_STOPPED, PPP_S_REQ_SENT, PPP_S_REQ_SENT, PPP_S_ACK_SENT, PPP_S_REQ_SENT },
+    { PPP_S_UNDEF, PPP_S_UNDEF, PPP_S_CLOSED, PPP_S_STOPPED, PPP_S_CLOSING, 
+        PPP_S_STOPPING, PPP_S_REQ_SENT, PPP_S_ACK_RCVD, PPP_S_ACK_SENT, PPP_S_OPENED },
+    { PPP_S_UNDEF, PPP_S_UNDEF, PPP_S_CLOSED, PPP_S_STOPPED, PPP_S_CLOSING, 
+        PPP_S_STOPPING, PPP_S_REQ_SENT, PPP_S_REQ_SENT, PPP_S_ACK_SENT, PPP_S_OPENED },
+    { PPP_S_UNDEF, PPP_S_UNDEF, PPP_S_CLOSED, PPP_S_STOPPED, PPP_S_CLOSED, 
+        PPP_S_STOPPED, PPP_S_STOPPED, PPP_S_STOPPED, PPP_S_STOPPED, PPP_S_STOPPING },
+    { PPP_S_UNDEF, PPP_S_UNDEF, PPP_S_CLOSED, PPP_S_STOPPED, PPP_S_CLOSING, 
+        PPP_S_STOPPING, PPP_S_REQ_SENT, PPP_S_ACK_RCVD, PPP_S_ACK_SENT, PPP_S_OPENED }
+};
+
+
+
+typedef struct gnrc_ppp_fsm gnrc_ppp_fsm_t;
+typedef struct gnrc_ppp_fsm_conf gnrc_ppp_fsm_conf_t;
+
+
+/**
+ * @brief Option Negotiation Automaton FSM actions
+ *
+ * @see https://tools.ietf.org/html/rfc1661#section-4.4
+ */
+static const uint16_t actions[PPP_NUM_EVENTS][PPP_NUM_STATES] = {
+    { 0, PPP_F_IRC | PPP_F_SCR, 0, 0, 0, 0, 0, 0, 0, 0 },
+    { 0, 0, 0, PPP_F_TLS, 0, 0, 0, 0, 0, PPP_F_TLD },
+    { PPP_F_TLS, 0, PPP_F_IRC | PPP_F_SCR, 0, 0, 0, 0, 0, 0, 0 },
+    { 0, PPP_F_TLF, 0, 0, 0, 0, PPP_F_IRC | PPP_F_STR, PPP_F_IRC |
+        PPP_F_STR, PPP_F_IRC | PPP_F_STR, PPP_F_TLD | PPP_F_IRC | PPP_F_STR },
+    { 0, 0, 0, 0, PPP_F_STR, PPP_F_STR, PPP_F_SCR, PPP_F_SCR, PPP_F_SCR, 0 },
+    { 0, 0, 0, 0, PPP_F_TLF, PPP_F_TLF, PPP_F_TLF, PPP_F_TLF, PPP_F_TLF, 0 },
+    { 0, 0, PPP_F_STA, PPP_F_IRC | PPP_F_SCR | PPP_F_SCA, 0, 0, PPP_F_SCA,
+        PPP_F_SCA | PPP_F_TLU, PPP_F_SCA, PPP_F_TLD | PPP_F_SCR | PPP_F_SCA },
+    { 0, 0, PPP_F_STA, PPP_F_IRC | PPP_F_SCR | PPP_F_SCN, 0, 0, PPP_F_SCN,
+        PPP_F_SCN, PPP_F_SCN, PPP_F_TLD | PPP_F_SCR | PPP_F_SCN },
+    { 0, 0, PPP_F_STA, PPP_F_STA, 0, 0, PPP_F_IRC, PPP_F_SCR, PPP_F_IRC |
+        PPP_F_TLU, PPP_F_TLD | PPP_F_SCR },
+    { 0, 0, PPP_F_STA, PPP_F_STA, 0, 0, PPP_F_IRC | PPP_F_SCR, PPP_F_SCR,
+        PPP_F_IRC | PPP_F_SCR, PPP_F_TLD | PPP_F_SCR },
+    { 0, 0, PPP_F_STA, PPP_F_STA, PPP_F_STA, PPP_F_STA, PPP_F_STA, PPP_F_STA,
+        PPP_F_STA, PPP_F_TLD | PPP_F_ZRC | PPP_F_STA },
+    { 0, 0, 0, 0, PPP_F_TLF, PPP_F_TLF, 0, 0, 0, PPP_F_TLD | PPP_F_SCR },
+    { 0, 0, PPP_F_SCJ, PPP_F_SCJ, PPP_F_SCJ, PPP_F_SCJ, PPP_F_SCJ, PPP_F_SCJ,
+        PPP_F_SCJ, PPP_F_SCJ },
+    { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 },
+    { 0, 0, PPP_F_TLF, PPP_F_TLF, PPP_F_TLF, PPP_F_TLF, PPP_F_TLF, PPP_F_TLF,
+        PPP_F_TLF, PPP_F_TLD | PPP_F_IRC | PPP_F_STR },
+    { 0, 0, 0, 0, 0, 0, 0, 0, 0, PPP_F_SER }
+};
+
+/**
+ * @brief Representation of the Option Negotiation Automaton FSM
+ *
+ * @extends ppp_protocol_t
+ *
+ */
+typedef struct gnrc_ppp_fsm {
```

Comment:
You `typedef`'d that already in [l. 157](https://github.com/RIOT-OS/RIOT/pull/5470/files#diff-03275254ed895051fa03d50bd166e1f4R157). Please only use

```
struct gnrc_ppp_fsm {
    ...
};
```

here.